service/src/main/java/bio/terra/workspace/service/resource/ResourceValidationUtils.java

```diff
@@ -288,300 +261,9 @@ private boolean hasValidHostName(String hostName) {
-    if (StringUtils.isEmpty(vmImage.getUri())
-        && StringUtils.isEmpty(vmImage.getPublisher())
-        && StringUtils.isEmpty(vmImage.getOffer())
-        && StringUtils.isEmpty(vmImage.getSku())
-        && StringUtils.isEmpty(vmImage.getVersion())) {
-      throw new MissingRequiredFieldException(
-          "Missing required fields for vmImage. Either uri or publisher, offer, sku, version should be defined.");
-    }
-    if (StringUtils.isEmpty(vmImage.getUri())
-        && (StringUtils.isEmpty(vmImage.getPublisher())
-            || StringUtils.isEmpty(vmImage.getOffer())
-            || StringUtils.isEmpty(vmImage.getSku())
-            || StringUtils.isEmpty(vmImage.getVersion()))) {
-      throw new MissingRequiredFieldException(
-          "Missing required fields for vmImage. Publisher, offer, sku, version should be defined.");
-    }
-    if (StringUtils.isEmpty(vmImage.getUri())
-        && (!StringUtils.isEmpty(vmImage.getPublisher())
-            && !StringUtils.isEmpty(vmImage.getOffer())
-            && !StringUtils.isEmpty(vmImage.getSku())
-            && !StringUtils.isEmpty(vmImage.getVersion()))) {
-      checkFieldNonNull(apiAzureVmCreationParameters.getVmUser(), "vmUser");
-    }
-  }
-
-  /**
-   * Assert that the cloning instructions specified for a controlled or referenced resource are a
-   * valid combination. Intended for use at the api controller level.
-   *
-   * @param stewardshipType - controlled or referenced
-   * @param cloningInstructions - supplied cloning instructions with the API request
-   * @throws ValidationException if the combination is not valid
-   */
-  public static void validateCloningInstructions(
-      StewardshipType stewardshipType, CloningInstructions cloningInstructions) {
-    switch (stewardshipType) {
-      case CONTROLLED:
-        if (cloningInstructions.isValidForControlledResource()) {
-          return;
-        }
-        break;
-      case REFERENCED:
-        if (cloningInstructions.isValidForReferencedResource()) {
-          return;
-        }
-    }
-    throw new ValidationException(
-        String.format(
-            "Cloning Instruction %s is not valid with Stewardship Type %s",
-            cloningInstructions.toString(), stewardshipType));
-  }
-
-  /**
-   * Validate the terra reserved properties has valid values.
-   *
-   * @param properties of a resource.
-   */
-  public static void validateProperties(Map<String, String> properties) {
-    if (properties.containsKey(FOLDER_ID_KEY)) {
-      try {
-        var unused = UUID.fromString(properties.get(FOLDER_ID_KEY));
-      } catch (IllegalArgumentException e) {
-        throw new BadRequestException(
-            String.format(
-                "Property %s contains an invalid non-UUID format folder id %s.",
-                FOLDER_ID_KEY, properties.get(FOLDER_ID_KEY)));
-      }
-    }
-  }
+  private static final int MAX_FLEXIBLE_RESOURCE_DATA_BYTE_SIZE = 5000;
```

Comment:
nit, while you are here, could you move those static field to the top of the file?